h2o-docs/src/product/data-science/decision-tree.rst

```diff
@@ -0,0 +1,145 @@
+
+	a. *Maximum tree depth*: The process stops when the tree reaches a predefined maximum depth (set with ``max_depth``).
+	b. *Minimum number of samples*: The process stops when a node contains a minimum number of samples or instances (set with ``min_rows``).
+	c. *Limit of entropy improvement*: The process stops when the improvement of entropy between the parent node and the current split reaches a specified minimum threshold (refer to the `Entropy as a Splitting Rule <#entropy-as-a-splitting-rule>`__ section for more information).
+
+6. **Leaf node labeling**: The leaf nodes are labeled with the majority class or predicted value based on the training samples they contain at the end of the tree construction.
+
+Binning for Tree Building
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To handle large datasets efficiently, the Decision Tree algorithm utilizes binning as a preprocessing step at each internal node. Binning involves discretizing continuous features into a finitie number of bins. This reduces the computational complexity of finding the best attribute and threshold for each split.
+
+The binning process can be summarized as:
+
+1. **Input**: Inputting continuous feature values for a particular attribute.
+2. **Binning algorithm**: Appling a binning algorithm to divide the feature values into a specified number of bins (this technique is equal width binning).
+3. **Binning result**: Obtaining the bin boundaries for the feature values.
+4. **Statistic calculation**: Calculating statistics for each bin, such as the count of samples for each class.
+
+The binned features are then used for split point selection during tree construction. This allows for faster computation.
+
+Entropy as a Splitting Rule
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Decision Tree algorithm employs entropy as the splitting rule to determine the best attribute and threshold for each node. Entropy measures the impurity or disorder within a set of samples. The goal is to find splits that minimize the entropy and create homogenous subsets with respect to the target variable.
+
+The entropy of a set :math:`S` with respect to a binary target variable can be calculated using the following forumla:
+
+.. math::
+	
+	\text{Entropy}(S) = -p_1 \times \log2 (p_1) - p_0 \times \log2(p_0)
+
+where
+
+- :math:`p_1` is the proportion of positive (or class 1) samples in :math:`S`
+- :math:`p_0` is the proportion of negative (or class 0) samples in :math:`S`
+
+The attribute and threshold combination that minimizes the weighted average of the entropies of the resulting subsets is selected as the best split point.
+
+Examples
+~~~~~~~~
+
+.. tabs::
+	.. code-tab:: r R
+
+		library(h2o)
+		h2o.init()
+
+		# Import the prostate dataset:
+		prostate <- h2o.importFile("http://s3.amazonaws.com/h2o-public-test-data/smalldata/prostate/prostate.csv")
+
+		# Set the target variable:
+		target_variable <- 'CAPSULE'
+		prostate[target_variable] <- as.factor(prostate['CAPSULE'])
+
+
```

Comment:
Test is incomplete:

# build the model
h2o_dt <- h2o.decision_tree(y = target_variable, training_frame = prostate, max_depth = depthLevel)
# predict with model
pred_frame <- h2o.predict(h2o_dt, prostate)